test/e2e_node/gpu_device_plugin.go

```diff
@@ -146,19 +150,28 @@ func restartKubelet(f *framework.Framework) {
 	kube := matches[0]
 	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kube)
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kube).CombinedOutput()
-	if err == nil {
-		return
+	if err != nil {
+		framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
 	}
-	framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
+	Eventually(func() bool {
+		afterSocks, err := filepath.Glob("/var/lib/kubelet/device-plugins/nvidiaGPU*.sock")
+		if err != nil || len(afterSocks) < 1 {
+			return false
+		}
+		framework.Logf("beforeSock %s afterSock %s numberOfGpus %d", beforeSocks[0], afterSocks[0], framework.NumberOfNVIDIAGPUs(getLocalNode(f)))
+		return (afterSocks[0] != beforeSocks[0] && framework.NumberOfNVIDIAGPUs(getLocalNode(f)) > 0)
```

Comment:
I'm not familiar with the details of GPUs. Why should the number of sockets change when the kubelet is restarted?

Reply:
We expect device plugin to re-register with kubelet through a new socket, so this check is to make sure re-registration happens.

Reply:
Ah, I misread this before. I'm not sure `Glob` is guaranteed to have a stable ordering, so you better sort slice before comparing the first. Alternatively:

```go
Eventually(func() ([]string, error) {
  return filepath.Glob("/var/lib/kubelet/device-plugins/nvidiaGPU*.sock")
}, 5*time.Minute, framework.Pall).ShouldNot(ConsistOf(beforeSocks))
```

Reply:
We actually only expect a single device plugin socket at a time, but the suggested version is really neat. Updated now. Thanks for the tip!